File: Data Structures & Algorithms/anagram-groups/submission-0.py
class Solution:
    def groupAnagrams(self, strs: list[str]) -> list[list[str]]:

        # make hashmap of sorted str
        # if the sorted str is in the hashamp, then append it to the list in that hashamp
        map = {}
        res = []

        for i in strs:
            if "".join(sorted(i)) not in map:
                map["".join(sorted(i))] = [i]
            else:
                map["".join(sorted(i))].append(i)

        for i in map.values():
            res.append(i)
        return res

File: Data Structures & Algorithms/daily-temperatures/submission-0.java
class Solution {
    public int[] dailyTemperatures(int[] temperatures) {
        int n = temperatures.length;
        int[] res = new int[n];
        for (int i = 0; i < n - 1; i++) {
            for (int j = i + 1; j < n; j++) {
                if (temperatures[j] > temperatures[i]) {
                    res[i] = j - i;
                    break;
                }
                res[i] = 0;
            }
        }
        res[n - 1] = 0; return res;
    }
}
